pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,879 @@
+        
+        time_limit : float
+            set the maximum blocking time within thermostream has to settle.
+            Script execution is blocked until either thermostream has settled
+            or time_limte has been exceeded.
+        
+        log : boolean
+            show polling progress in command line
+            
+        Returns
+        -------
+        none
+
+        Examples
+        -------
+        
+        configure a default force temperature operation and sweep to 100 degC.
+        Stop script execution until temperature has settled or if it runs longer
+        than 150 s and use logging
+        
+        >>> ts.conf_set_temp()
+        >>> ts.set_temp_wait_settling(100,150)
+        temp_set= 100.0 deg, temp = 27.2 deg, time elapsed =   10 s, status = not at temperature             
+        """
+        # delay (thermostream firmware issue)
+        # force set_temp
+        time.sleep(1)
+        t = 0
+        while( not self.at_temperature()): #assert at temperature
+            time.sleep(1)
+            t += 1
+            if log == True:
+                print(f"temp_set= {self.temperature_setpoint:4.1f} deg,",
+                      f"temp = {self.temperature:4.1f} deg,",
+                      f"time elapsed = {t:4.0f} s,",
+                      f"status = {self.temperature_event_status}")
+                
+            if t > time_limit:
+                print('no settling achieved')
+                break
+
+    def stop(self):
+        # shut down thermo stream
+        self.enable_air_flow = 0
+        self.remote_mode = False
+        self.head = 'up'
+
+    def start(self):
+        self.remote_mode = True
+        self.enable_air_flow = 1 # enable thermostream
+
+    def cycling_stopped(self):
+        """check if cycling has stopped.      
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.CYCLING_STOPPED
+
+    def end_of_all_cycles(self):
+        """ check if cycling has stopped.
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.END_OF_ALL_CYCLES
+
+    def end_of_one_cycle(self):
+        """ check if end of (at least) one cycle is done.
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.END_OF_ONE_CYCLE
+
+    def end_of_test(self):
+        """ check if end of test is reached.
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.END_OF_TEST
+
+    def not_at_temperature(self):
+        """ check if not (yet) at temperature.
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.NOT_AT_TEMPERATURE
+
+    def at_temperature(self):
+        """ check if at temperature.
+        """
+        return self.temperature_condition_status_code & TemperatureStatusCode.AT_TEMPERATURE
+    
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
+        self.stop()
```

Comment:
This is a neat idea! I would prefer if you used/override the `shutdown()` method [defined](https://github.com/pymeasure/pymeasure/blob/e340f1c43dc302badbea46f060967199d508ec95/pymeasure/instruments/instrument.py#L303-L306) in `Instrument`.

Reply:
I updated the .stop() method